runtime/vm/CRIUHelpers.cpp

```diff
@@ -389,4 +402,70 @@ runInternalJVMRestoreHooks(J9VMThread *currentThread)
 	return result;
 }
 
+BOOLEAN
+runDelayedIdentityOperations(J9VMThread *currentThread)
+{
+	pool_state walkState = {0};
+	J9JavaVM *vm = currentThread->javaVM;
+	J9Pool *delayedSingleThreadModeOpRecords = vm->checkpointState.delayedSingleThreadModeOpRecord;
+	J9InternalVMFunctions* vmFuncs = vm->internalVMFunctions;
+	J9DelayedSingleThreadModeOpRecord *delayedSingleThreadModeOp = (J9DelayedSingleThreadModeOpRecord*) pool_startDo(delayedSingleThreadModeOpRecords, &walkState);
+	omrthread_monitor_t monitorPtr = NULL;
+	BOOLEAN rc = TRUE;
+
+	while (NULL != delayedSingleThreadModeOp) {
+		j9object_t instance = J9_JNI_UNWRAP_REFERENCE(delayedSingleThreadModeOp->globalObjectRef);
+		if (!VM_ObjectMonitor::inlineFastObjectMonitorEnter(currentThread, instance)) {
+			rc = objectMonitorEnterNonBlocking(currentThread, instance);
+			if (J9_OBJECT_MONITOR_BLOCKING == rc) {
+				/* owned by another thread */
+				rc = FALSE;
+				goto done;
```

Comment:
This should get a tracepoint to indicate which monitor caused the problem.  And to make life really easy, I'd add a field to the checkpoint struct for `J9DelayedSingleThreadModeOpRecord failedMonitorOp` and store the pointer there to make servicing this easier